### PR TITLE
Updated Matrix_Expo.mdx with python code

### PR DESCRIPTION
**Pull request by 19WH1A0524**
This is a cses program code.

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/5_Plat/Matrix_Expo.mdx [hunk 1]
```diff
@@ -124,7 +128,48 @@ int main() {
 	cout << base[0][1];
 }
 ```
+</CPPSection>
+
+<PySection>
+```python
+import sys
```
Comment (envyaims) [inline content/5_Plat/Matrix_Expo.mdx:135]: you didnt even use it

File: content/5_Plat/Matrix_Expo.mdx [hunk 2]
```diff
@@ -124,7 +128,48 @@ int main() {
...
+MOD = 1000000007
+
+def mul(A, B, MOD):
+    C = [[0,0], [0,0]]
+
+    for i in range(2):
+        for j in range(2):
+            for k in range(2):
+                C[i][j] += ((A[i][k] % MOD) * (B[k][j] % MOD)) % MOD
```
Comment (envyaims) [inline content/5_Plat/Matrix_Expo.mdx:144]: not necessary for so many mods? just one at the end is fine since the previous values are already modded

**Review by envyaims — CHANGES_REQUESTED** [review]

File: content/5_Plat/Matrix_Expo.mdx [hunk 3]
```diff
@@ -124,7 +128,48 @@ int main() {
...
+
+    return C
 
+def Power(m, n, MOD):
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:148]: rename to `power`

File: content/5_Plat/Matrix_Expo.mdx [hunk 4]
```diff
@@ -124,7 +128,48 @@ int main() {
...
+	if n == 0:
+		return [[1,0], [0,1]]
+	if n == 1:
+		return m
+
+	hf = Power(m, n // 2, MOD)
+  	sq = mul(hf, hf, MOD)
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:155]: what do `hf` and `sq` stand for

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/5_Plat/Matrix_Expo.mdx (hunk 4 above)
Comment (envyaims) [reply content/5_Plat/Matrix_Expo.mdx:155]: maybe write out `half` and `square`

File: content/5_Plat/Matrix_Expo.mdx [hunk 5]
```diff
@@ -124,7 +128,48 @@ int main() {
...
+	if n % 2 == 0:
+		return sq
+
+	return mul(m, sq, MOD)
+
+
+n = int(input())
+if n > 1:
+	o = Power([[1,1], [1,0]], n, MOD)
+        #print(o)
```
Comment (ryanchou-dev) [inline content/5_Plat/Matrix_Expo.mdx:166]: remove

File: content/5_Plat/Matrix_Expo.mdx [hunk 6]
```diff
@@ -124,7 +128,48 @@ int main() {
...
+    for i in range(2):
+        for j in range(2):
+            for k in range(2):
+                C[i][j] += ((A[i][k] % MOD) * (B[k][j] % MOD)) % MOD
+
+    return C
 
+def Power(m, n, MOD):
+	if n == 0:
+		return [[1,0], [0,1]]
```
Comment (ryanchou-dev) [inline content/5_Plat/Matrix_Expo.mdx:150]: space out ops

File: content/5_Plat/Matrix_Expo.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply content/5_Plat/Matrix_Expo.mdx:144]: ^^

File: content/5_Plat/Matrix_Expo.mdx [hunk 7]
```diff
@@ -124,7 +128,48 @@ int main() {
...
 ```
+</CPPSection>
+
+<PySection>
+```python
+import sys
+MOD = 1000000007
+
+def mul(A, B, MOD):
+    C = [[0,0], [0,0]]
```
Comment (ryanchou-dev) [inline content/5_Plat/Matrix_Expo.mdx:139]: here too

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]
Once the above are fixed I'll add my comments

**Comment by 19WH1A0524** [comment]
I will make the pull request, I guess I have missed the mail. Sorry about
that.

File: content/5_Plat/Matrix_Expo.mdx (hunk 2 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx]: made the changes

File: content/5_Plat/Matrix_Expo.mdx (hunk 3 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx]: made the changes

File: content/5_Plat/Matrix_Expo.mdx (hunk 4 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx]: made the changes

File: content/5_Plat/Matrix_Expo.mdx (hunk 6 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx]: made the changes

File: content/5_Plat/Matrix_Expo.mdx [hunk 8]
```diff
@@ -124,7 +128,46 @@ int main() {
...
 }
 ```
+</CPPSection>
+
+<PySection>
+```python
+
+MOD = 1000000007
+
+def mul(A,B,MOD):
```
Comment (SansPapyrus683) [inline content/5_Plat/Matrix_Expo.mdx:138]: type annoations + space stuff out

**Comment by 19WH1A0524** [comment]
done.

**Comment by SansPapyrus683** [comment]
you literally haven't committed anything

**Comment by 19WH1A0524** [comment]
I did the spacing

**Comment by SansPapyrus683** [comment]
? i don't see it?

**Comment by 19WH1A0524** [comment]
> Once the above are fixed I'll add my comments

 I made all the changes

**Comment by 19WH1A0524** [comment]
Are there any more changes to be made?

**Comment by SansPapyrus683** [comment]
@ryanchou-dev

File: content/5_Plat/Matrix_Expo.mdx [hunk 9]
```diff
@@ -124,7 +128,46 @@ int main() {
...
+def power(matrix : List[List[int]], n : int) -> List[List[int]]:
+	if n == 0:
+        	return [[1, 0], [0, 1]]
+    	if n == 1:
+        	return matrix
+    	half = power(matrix, n // 2, MOD)
+    	square = mul(half, half, MOD)
+    	if n % 2 == 0:
+        	return square
+    	return mul(matrix, square, MOD)
```
Comment (ryanchou-dev) [inline content/5_Plat/Matrix_Expo.mdx:156]: what happened to the indentation here?

**Review by ryanchou-dev — APPROVED** [review]
other than that, all good ⭐

File: content/5_Plat/Matrix_Expo.mdx (hunk 9 above)
Comment (19WH1A0524) [reply content/5_Plat/Matrix_Expo.mdx]: I dint actually change anything, committed now please check again

**Comment by 19WH1A0524** [comment]
@SansPapyrus683

**Comment by 19WH1A0524** [comment]
Hey, I forgot to add my name, can i make another pr ?

**Pull request merged**